DXRMapFixups/DeusEx/Classes/DXRFixupVandenberg.uc

```diff
@@ -704,20 +725,69 @@ function TimerMapFixes()
         break;
 
     case "14_Oceanlab_silo":
-        prevMapsDone = dxr.flagbase.GetBool('Heliosborn') &&  //Finished Vandenberg, mission 12
-            dxr.flagbase.GetBool('schematic_downloaded'); //Finished Ocean Lab, mission 14,
-        prevMapsDone = prevMapsDone || !#defined(injections) || dxr.flags.settings.goals<=0;
+        // by design, no infolink plays after killing Howard Strong if the missile hasn't been redirected
+        _MissileLaunchedChecks();
+        _CheckMoreSiloInfolinks();
+        break;
+    }
+}
 
-        if(prevMapsDone
-           && dxr.flagbase.GetBool('missile_launched') //Redirected the missile in Silo, mission 14
-           && !dxr.flagbase.GetBool('MS_HowardStrongUnhidden'))
-        {
-            foreach AllActors(class'#var(prefix)HowardStrong', hs) {
-                hs.EnterWorld();
+function private _MissileLaunchedChecks() {
+    local #var(prefix)HowardStrong hs;
+    local bool strongAlive;
+    local BlackHelicopter chopper;
+
+    if (dxr.flagbase.GetBool('DXR_SiloEscapeHelicopterUnhidden') || !dxr.flagbase.GetBool('missile_launched')) {
+        return;
+    }
+
+    foreach AllActors(class'HowardStrong', hs) {
+        strongAlive = True;
+        break;
+    }
+
+    if (strongAlive == False) {
+        // both goals completed just now
+        foreach AllActors(class'BlackHelicopter', chopper, 'BlackHelicopter') {
+            chopper.EnterWorld();
+            break;
+        }
+        dxr.flagbase.SetBool('DXR_SiloEscapeHelicopterUnhidden', True,, 15);
+    } else {
+        // only computer goal completed
+        player().StartDataLinkTransmission("DL_Savage3");
+    }
+}
+
+function private _CheckMoreSiloInfolinks() {
+    local ConEvent ce;
+
+    if (!dxr.flagbase.GetBool('DXR_SiloEscapeHelicopterUnhidden')) {
+        return;
+    }
+
+    if (dxr.flagbase.GetBool('DL_Savage3_Played')) {
+        // both goals completed, computer infolink played
+        player().StartDataLinkTransmission("DL_Dead");
+    } else {
+        // both goals completed, computer infolink not played
+
+        for (ce = GetConversation('DL_Savage3').eventList; ce != None; ce = ce.nextEvent) {
+            if (ce.nextEvent != None && ce.nextEvent.eventType == ET_Speech) {
+                if (
+                    ConEventSpeech(ce.nextEvent).conSpeech.speech == "Just like your lab work, Savage: premature celebration." ||
+                    ConEventSpeech(ce.nextEvent).conSpeech.speech == "As a scientist, you should have learned to expect surprises."
+                ) {
+                    ce.nextEvent = ce.nextEvent.nextEvent;
```

Comment:
couldn't you just stitch the line before "premature celebration" straight to `DL_Dead`? I don't think we need to look for or worry about "expect surprises"

Reply:
oh right, you're looking to keep "you've got 10 minutes"

Reply:
idk if it would be easier to search for the lines we want to keep instead of the lines we want to exclude lol

Reply:
Do you have a solution in mind that doesn't require multiple loops?

Reply:
I guess you could still do it with just one pass over the conversation. Might not be quite as clean though?

Reply:
We definitely only need a single pass through the conversation for this, since only one line needs to be cut, and the other conversation stitched in.

Reply:
well I do like keeping the "you have 10 minutes part", so there is a bit of a gap

still can do it with a couple of well named local variables

Reply:
Only the "premature celebration" line needs to be explicitly cut and stitched out.  After the "10 minutes" line, you can directly connect to the other conversation
![image](https://github.com/Die4Ever/deus-ex-randomizer/assets/13684088/3c8cc61d-fd21-48c3-b4a3-bf9430593ae1)